### ISSUE TITLE
Issue with captcha

### ISSUE THREAD

**Issue by NelsonLombo** [xlrl/docker-mantisbt#1]
I downloaded the DockerHub image and installed it in my Docker cluster (Engine version --> 18.03.0-ce) I have done the configuration of the php home page following the steps described in Dockerhub, but when creating a new user in Mantis, captcha does not work, so it is not possible to create users on the platform. I've followed the same steps you indicate, I don't know if I'm the only one who has the problem or nobody has notified before. Could you help me?

Thank you in advance,
![captcha](https://user-images.githubusercontent.com/43599825/47294375-2f393f80-d60d-11e8-8b80-f2798224d7a7.JPG)

Greetings

**Comment by XelaRellum** [comment]
As we don't allow self-service user creation in our own Mantis instances
we never had to use captchas before.

I just tried to see if this is easily fixable, which it was not. At least, wasn't solvable using the results from the Mantis forum.

The image still runs on PHP5 so this needs changing anyway and maybe moving to alpine/php7 fixes this problem. Feel free to try working on this and I will happily integrate this. Hope to find time to do this...

**Comment by FrancYescO** [comment]
fixed issuing thse commands, maybe embed in the dockerfile:
https://stackoverflow.com/a/26712001

**Comment by XelaRellum** [comment]
@FrancYescO  thx a lot for the information. I have included this. Can you try out the latest image (not tagged yet) and tell me if this has worked for you?

**Comment by FrancYescO** [comment]
i think due to the release of php 8.1, the image breaks:
```
Deprecated: Return type of ADODB_Iterator_empty::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3174

Deprecated: Return type of ADODB_Iterator_empty::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3178

Deprecated: Return type of ADODB_Iterator_empty::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3170

Deprecated: Return type of ADODB_Iterator_empty::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3166

Deprecated: Return type of ADODB_Iterator_empty::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3164

Deprecated: Return type of ADORecordSet_empty::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3229

Deprecated: Return type of ADODB_Iterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3301

Deprecated: Return type of ADODB_Iterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3305

Deprecated: Return type of ADODB_Iterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3297

Deprecated: Return type of ADODB_Iterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3293

Deprecated: Return type of ADODB_Iterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3289

Deprecated: Return type of ADORecordSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3381

Deprecated: Constant FILTER_SANITIZE_STRING is deprecated in /var/www/html/config_defaults_inc.php on line 219
Possible Whitespace/Error in Configuration File - Aborting. Output so far follows:
string(3874) "
Deprecated: Return type of ADODB_Iterator_empty::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3174

Deprecated: Return type of ADODB_Iterator_empty::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3178

Deprecated: Return type of ADODB_Iterator_empty::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3170

Deprecated: Return type of ADODB_Iterator_empty::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3166

Deprecated: Return type of ADODB_Iterator_empty::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3164

Deprecated: Return type of ADORecordSet_empty::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3229

Deprecated: Return type of ADODB_Iterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3301

Deprecated: Return type of ADODB_Iterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3305

Deprecated: Return type of ADODB_Iterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3297

Deprecated: Return type of ADODB_Iterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3293

Deprecated: Return type of ADODB_Iterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3289

Deprecated: Return type of ADORecordSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3381

Deprecated: Constant FILTER_SANITIZE_STRING is deprecated in /var/www/html/config_defaults_inc.php on line 219
"
```

i've solved using `php:8.0-apache` instead of `php:apache` as starting image.

btw, please consider somehow to embed also https://github.com/xlrl/docker-mantisbt/pull/12 to avoid that on each update of the container we need to redisable the admin dir (you can also just check if a config exist and than chmod 000 the dir)

just as another suggestion: having the possibility to set SMTP, LDAP (and MYSQL) configs via ENV will allow to have out of the box ready to work coinatainer skipping the needs of first config via webpage or editing the config volumes (volumes will became useless probably)

**Comment by XelaRellum** [comment]
> i think due to the release of php 8.1, the image breaks:
> 
> ```
> Deprecated: Return type of ADODB_Iterator_empty::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3174
> 
> Deprecated: Return type of ADODB_Iterator_empty::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3178
> 
> Deprecated: Return type of ADODB_Iterator_empty::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3170
> 
> Deprecated: Return type of ADODB_Iterator_empty::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3166
> 
> Deprecated: Return type of ADODB_Iterator_empty::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3164
> 
> Deprecated: Return type of ADORecordSet_empty::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3229
> 
> Deprecated: Return type of ADODB_Iterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3301
> 
> Deprecated: Return type of ADODB_Iterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3305
> 
> Deprecated: Return type of ADODB_Iterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3297
> 
> Deprecated: Return type of ADODB_Iterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3293
> 
> Deprecated: Return type of ADODB_Iterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3289
> 
> Deprecated: Return type of ADORecordSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3381
> 
> Deprecated: Constant FILTER_SANITIZE_STRING is deprecated in /var/www/html/config_defaults_inc.php on line 219
> Possible Whitespace/Error in Configuration File - Aborting. Output so far follows:
> string(3874) "
> Deprecated: Return type of ADODB_Iterator_empty::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3174
> 
> Deprecated: Return type of ADODB_Iterator_empty::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3178
> 
> Deprecated: Return type of ADODB_Iterator_empty::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3170
> 
> Deprecated: Return type of ADODB_Iterator_empty::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3166
> 
> Deprecated: Return type of ADODB_Iterator_empty::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3164
> 
> Deprecated: Return type of ADORecordSet_empty::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3229
> 
> Deprecated: Return type of ADODB_Iterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3301
> 
> Deprecated: Return type of ADODB_Iterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3305
> 
> Deprecated: Return type of ADODB_Iterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3297
> 
> Deprecated: Return type of ADODB_Iterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3293
> 
> Deprecated: Return type of ADODB_Iterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3289
> 
> Deprecated: Return type of ADORecordSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/adodb/adodb-php/adodb.inc.php on line 3381
> 
> Deprecated: Constant FILTER_SANITIZE_STRING is deprecated in /var/www/html/config_defaults_inc.php on line 219
> "
> ```
> 
> i've solved using `php:8.0-apache` instead of `php:apache` as starting image.
> 
> btw, please consider somehow to embed also #12 to avoid that on each update of the container we need to redisable the admin dir (you can also just check if a config exist and than chmod 000 the dir)
> 
> just as another suggestion: having the possibility to set SMTP, LDAP (and MYSQL) configs via ENV will allow to have out of the box ready to work coinatainer skipping the needs of first config via webpage or editing the config volumes (volumes will became useless probably)

Yea, take a look at #15. Should be fixed now. Can you check if the captchas no work for you?

**Comment by FrancYescO** [comment]
as said i've tried with php 8.0 and captcha was ok with the commands you added in the untagged release

**Comment by XelaRellum** [comment]
Ok, then I am going to close the issue.

**Comment by FrancYescO** [comment]
hmm @XelaRellum seems latest tagged image on dockerhub does not have this fix

**Comment by XelaRellum** [comment]
> hmm @XelaRellum seems latest tagged image on dockerhub does not have this fix

You are right. The fix was in a different branch.
Can you check the image tagged with 2.25.2-captcha?

**Comment by XelaRellum** [comment]
I have setup a test server did a quick check if 2.25.2-captcha works. To me it seems that way, so I updated the tags 2.25.2 and latest

**Comment by FrancYescO** [comment]
Hello, looking at the [latest changelog](https://www.mantisbt.org/bugs/changelog_page.php?project=mantisbt&version=2.25.6) seems they've fixed the issue of captcha with php 8.1 maybe you can consider switching back to `php:apache`
